test/test_autograd.py

```diff
@@ -2236,36 +2236,6 @@ class dont_convert(tuple):
     ('det', lambda: random_fullrank_matrix_distinct_singular_value(S), (), 'distinct_postive_s', (), [skipIfNoLapack]),
     ('svd', lambda: random_fullrank_matrix_distinct_singular_value(S), (), '', (), [skipIfNoLapack]),
     ('gesv', (S, S), ((S, S),), '', (), [skipIfNoLapack]),
-    ('eq', (S, S, S), ((S, S, S),)),
```

Comment:
How did you check that these methods are still being tested?

Reply:
but these did test some useful stuff like that inplace worked; we are just throwing that away?  Can't we just skip the specific part of the check that triggers the error if the return type of the function is non-differentiable?

Reply:
We should fix this, but I'm not sure how yet. I don't think we should ignore outputs that don't require_grad. I think we should be checking that the gradient is actually zero. Otherwise we miss things that should have gradient definitions, but are erroneously marked as not requiring grad.

Reply:
I agree.  I just meant for the purposes of this PR you can leave the useful checks and avoid the error by looking at the return type of the function (not if the function requires grad or not).

Reply:
I added back the working in-place tests (`broadcast_rhs/all` doesn't work with in-place).

Adding an exclude requires excluding all of `do_test`, so it won't actually do anything.

Reply:
that seems fine, but there should have already been guards against running broadcast_lhs/broadcast_all tests inplace, so I don't know why you had to do this.